plugins/vpp/ipsecplugin/vppcalls/dump_vppcalls.go

```diff
@@ -0,0 +1,385 @@
+			saGenName := "sa-generated-" + strconv.Itoa(paIdx)
+			paIdx++
+
+			// Addresses
+			var remoteStartAddrStr, remoteStopAddrStr, localStartAddrStr, localStopAddrStr string
+			if uintToBool(spdDetails.IsIpv6) {
+				var remoteStartAddr, remoteStopAddr net.IP = spdDetails.RemoteStartAddr, spdDetails.RemoteStopAddr
+				remoteStartAddrStr, remoteStopAddrStr = remoteStartAddr.String(), remoteStopAddr.String()
+				var localStartAddr, localStopAddr net.IP = spdDetails.LocalStartAddr, spdDetails.LocalStopAddr
+				localStartAddrStr, localStopAddrStr = localStartAddr.String(), localStopAddr.String()
+			} else {
+				var remoteStartAddr, remoteStopAddr net.IP = spdDetails.RemoteStartAddr[:4], spdDetails.RemoteStopAddr[:4]
+				remoteStartAddrStr, remoteStopAddrStr = remoteStartAddr.String(), remoteStopAddr.String()
+				var localStartAddr, localStopAddr net.IP = spdDetails.LocalStartAddr[:4], spdDetails.LocalStopAddr[:4]
+				localStartAddrStr, localStopAddrStr = localStartAddr.String(), localStopAddr.String()
+			}
+
+			// Prepare policy entry and put to the SPD
+			policyEntry := &ipsec.SecurityPolicyDatabases_SPD_PolicyEntry{
+				Sa:              saGenName,
+				Priority:        spdDetails.Priority,
+				IsOutbound:      uintToBool(spdDetails.IsOutbound),
+				RemoteAddrStart: remoteStartAddrStr,
+				RemoteAddrStop:  remoteStopAddrStr,
+				LocalAddrStart:  localStartAddrStr,
+				LocalAddrStop:   localStopAddrStr,
+				Protocol:        uint32(spdDetails.Protocol),
+				RemotePortStart: uint32(spdDetails.RemoteStartPort),
+				RemotePortStop:  uint32(spdDetails.RemoteStopPort),
+				LocalPortStart:  uint32(spdDetails.LocalStartPort),
+				LocalPortStop:   uint32(spdDetails.LocalStopPort),
+			}
+			spd.PolicyEntries = append(spd.PolicyEntries, policyEntry)
+
+			// Prepare meta and put to the metadata map
+			meta := &SpdMeta{
+				SpdID:   spdDetails.SpdID,
+				SaID:    spdDetails.SaID,
+				Policy:  spdDetails.Policy,
+				Bytes:   spdDetails.Bytes,
+				Packets: spdDetails.Packets,
+			}
+			metadata.SpdMeta[saGenName] = meta
+		}
+		// Store STD in list
+		spdList = append(spdList, &IPSecSpdDetails{
+			Spd:  spd,
+			Meta: metadata,
+		})
+	}
+
+	return spdList, nil
+}
+
+// Get all security association (used also for tunnel interfaces) in binary api format
+func (handler *ipSecVppHandler) dumpSecurityAssociations(saID uint32) (saList []*ipsecapi.IpsecSaDetails, err error) {
+	defer func(t time.Time) {
+		handler.stopwatch.TimeLog(ipsecapi.IpsecSaDump{}).LogTimeEntry(time.Since(t))
+	}(time.Now())
+
+	req := &ipsecapi.IpsecSaDump{
+		SaID: saID,
+	}
+	requestCtx := handler.callsChannel.SendMultiRequest(req)
+
+	for {
+		saDetails := &ipsecapi.IpsecSaDetails{}
+		stop, err := requestCtx.ReceiveReply(saDetails)
+		if stop {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		saList = append(saList, saDetails)
+	}
+
+	return saList, nil
+
+}
+
+func getCryptoAlg(alg uint8) ipsec.CryptoAlgorithm {
```

Comment:
This is unnecessary the value from binapi can be converted to `CryptoAlgorithm` type directly, no need to have another place of conversion.

For configuration it's also just converted directly: https://github.com/ligato/vpp-agent/blob/ea8e83b4c762a466bf1dbde1638a7cb34fe8e229/plugins/vpp/ipsecplugin/vppcalls/ipsec_vppcalls.go#L81